modules/document_repository/php/document_repository.class.inc

```diff
@@ -316,10 +316,15 @@ class Document_Repository extends \NDB_Menu_Filter
         foreach ($dbresult as $row) {
             $results[] =$row;
             $tree      = null;
+            // FIXME the for-loop below is pretty confusing and is not 
+            // documented. This should be cleaned up and the phan suppressions
+            // removed.
             foreach ($results as $result) {
+                //@phan-suppress-next-line PhanUndeclaredVariable
                 $thisref = &$refs->{$result['id']};
```

Comment:
black magic PHP reference juggling i just don't know

Reply:
Might be $ref was used before but some code modification removed it's declaration or other strange thing that append when someone modify some existing core.

Reply:
@PapillonMcGill no, as far as I know this code was written by someone before we had much of a code review process in place and was just always poorly written..

Reply:
@PapillonMcGill Yes I tried selectively removing parts of the could or declaring `$ref` and `$thisref` explicitly and it broke the tree structure.

Reply:
I mean remove from line 313 to 335